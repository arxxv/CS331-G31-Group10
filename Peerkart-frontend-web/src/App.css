.app {
	background: #f5f5f5;
	display: flex;
	width: 100vw;
	height: auto;
}

.app-left {
	width: 20%;
	height: 100%;
}

.app-right {
	width: 80%;
	height: 100%;
}
