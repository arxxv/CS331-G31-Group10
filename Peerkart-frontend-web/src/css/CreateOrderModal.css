.modal-cont {
	width: 100vw;
	height: 100vh;
	background: rgba(0, 0, 0, 0.5);
	position: fixed;
	top: 0;
	left: 0;
	display: flex;
	align-items: center;
	justify-content: center;
}

.modal-view {
	background-color: white;
	height: 80%;
	width: 60%;
	box-shadow: 0px 4px 6px -1px rgba(203, 12, 159, 0.12),
		0px 2px 4px -1px rgba(203, 12, 159, 0.07);
	border-radius: 8px;
}

.create-head {
	text-align: center;
	font-size: 1.5rem;
	font-weight: 800;
	font-family: poppins;
}

.create-inputs {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	width: 70%;
	padding: 1%;
	text-align: center;
}

.create-label {
	font-family: "Open Sans";
	padding: 1rem 0px;
	font-size: 1rem;
	font-weight: 600;
	text-transform: uppercase;
}

.create-label-itn {
	font-family: "Open Sans";
	padding: 0.7rem 0px;
	font-size: 1rem;
	font-weight: 600;
	text-transform: uppercase;
}

.create-label-unit {
	font-family: "Open Sans";
	padding: 5px 0px 10px;
	margin-top: -8px;
	font-size: 16px;
	font-weight: 600;
	text-transform: uppercase;
}

.create-input {
	padding: 0.1rem 5px;
	display: flex;
	flex-direction: column;
	/* background-color: rgb(232, 240, 254); */
	/* border: none; */
	/* outline: none; */
	height: 25px;
	border-width: 1px;
	border-color: rgb(133, 133, 133, 0.5);
	border-radius: 5px;
	font-size: 18px;
	width: 97%;
	margin: 10px 0px;
	text-align: center;
}

.create-input-qty {
	padding: 5px 5px;
	display: flex;
	flex-direction: column;
	border-width: 1px;
	border-color: rgb(133, 133, 133, 0.5);
	border-radius: 5px;
	outline: none;
	height: 25px;
	width: 50%;
	font-size: 18px;
	margin: 10px 0px;
	text-align: center;
}

.create-box {
	width: 100%;
	height: 100%;
	display: flex;
}

.createOrder-left {
	width: 55%;
	display: flex;
	justify-content: center;
}

.createOrder-right {
	width: 45%;
	height: 100%;
	display: flex;
	flex-direction: column;
}

.qty-unit {
	display: flex;
	width: 100%;
	justify-content: space-evenly;
	align-items: center;
	padding-right: 10%;
}

.unit {
	display: flex;
	flex-direction: column;
}

.add-item-btn {
	background: #ffffff;
	box-shadow: 0px 4px 6px -1px rgba(203, 12, 159, 0.12),
		0px 2px 4px -1px rgba(203, 12, 159, 0.07);
	border-radius: 8px;
	/* border: 1px solid rgb(203, 12, 159); */
	width: 120px;
	display: flex;
	align-items: center;
	justify-content: center;
	font-family: "Poppins", sans-serif;
	font-weight: 700;
	text-transform: uppercase;
	color: black;
	padding: 3% 7%;
	cursor: pointer;
	margin-top: 5%;
	margin-bottom: 5%;
}

.back {
	background: #ffffff;
	box-shadow: 0px 4px 6px -1px rgba(203, 12, 159, 0.12),
		0px 2px 4px -1px rgba(203, 12, 159, 0.07);
	border-radius: 8px;
	/* border: 1px solid rgb(203, 12, 159); */
	/* height: 100%; */
	width: 120px;
	margin-top: 5%;
	display: flex;
	align-items: center;
	justify-content: center;
	font-family: "Poppins", sans-serif;
	font-weight: 700;
	text-transform: uppercase;
	color: #000;
	padding: 3% 7%;
	cursor: pointer;
}

.btns {
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	padding: 30px 0;
}

.input-box {
	width: 100%;
	padding: 2% 0px;
}

.input-box-cat {
	margin-bottom: 10px;
	width: 100%;
}

.input-box-itn {
	width: 100%;
	padding: 2px 0px;
}

.qty-input-div {
	width: 100%;
	display: flex;
	align-items: center;
	justify-content: center;
}

.MuiSelect-outlined {
	/* padding: 10px !important; */
	align-items: center !important;
}

.units-qty {
	height: 45px;
}

.create-order-hr {
	height: 1px;
	color: #e6e6e6;
	width: 66em;
	width: 100%;
}

.label-right {
	font-family: "Poppins";
	font-weight: 700;
	font-size: 1rem;
}

.label-right-value {
	font-family: "Lora";
}

.checkout-btn {
	background: #ffffff;
	box-shadow: 0px 4px 6px -1px rgba(203, 12, 159, 0.12),
		0px 2px 4px -1px rgba(203, 12, 159, 0.07);
	border-radius: 8px;
	width: 120px;
	margin-top: 5%;
	display: flex;
	align-items: center;
	justify-content: center;
	font-family: "Poppins", sans-serif;
	font-weight: 700;
	text-transform: uppercase;
	color: #000;
	padding: 3% 7%;
	cursor: pointer;
}

.right-heads {
	height: 7%;
	padding: 1.25rem 0;
}

.right-heads-cat {
	height: 7%;
}

.items-wrapper {
	height: 45%;
}

.addCart-btn {
	height: 15%;
	display: flex;
	justify-content: center;
}

.item-row {
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding: 2% 5%;
}

.item-row-right {
	display: flex;
}

.item-cont {
	overflow-y: scroll;
	height: 80%;
}
