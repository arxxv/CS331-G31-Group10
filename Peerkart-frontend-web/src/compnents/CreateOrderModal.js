import React, { useState } from "react";
import "../css/CreateOrderModal.css";
import Box from "@mui/material/Box";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import LineImg from "../images/line.svg";
import { useDispatch, useSelector } from "react-redux";
import { addToCart } from "../redux/actions/cartActions";
import { Edit, Delete } from "@mui/icons-material";

const CreateOrderModal = ({ modalVisible, setModalVisible }) => {
	const [orderName, setOrderName] = useState("");
	const [name, setname] = useState("");
	const [category, setCategory] = useState("");
	const [qty, setQty] = useState("");
	const [unit, setUnit] = useState("");
	const dispatch = useDispatch();
	const itemsInCart = useSelector((state) => state.cart.items);

	return (
		<>
			{modalVisible ? (
				<div className='modal-cont'>
					<div className='modal-view'>
						<div className='create-head'>CREATE ORDER</div>
						<div className='create-box'>
							<div className='createOrder-left'>
								<div className='create-inputs'>
									<div className='input-box'>
										<label className='create-label'>Order Name</label>
										<input
											className='create-input'
											value={orderName}
											type='text'
											name='orderName'
											onChange={(e) => setOrderName(e.target.value)}
											placeholder='Eg. Daily Needs List'
										/>
									</div>
									<div className='input-box-cat'>
										{/* <label className='create-label'>Category</label>
										<select value={category}>
											<option value='Grocery'>Grocery</option>
											<option value='Fish ans meat'>Fish ans meat</option>
											<option value='Stationary'>Stationary</option>
										</select> */}
										<Box sx={{ minWidth: 120 }}>
											<FormControl fullWidth size='small' margin='none'>
												<InputLabel
													id='demo-simple-select-label'
													style={{ padding: "1px" }}>
													Category
												</InputLabel>
												<Select
													labelId='demo-simple-select-label'
													id='demo-simple-select'
													value={category}
													label='category'
													onChange={(e) => setCategory(e.target.value)}>
													<MenuItem value={"Groceries"}>Groceries</MenuItem>
													<MenuItem value={"Fish and Meat"}>
														Fish and Meat
													</MenuItem>
													<MenuItem value={"Stationary"}>Stationary</MenuItem>
													<MenuItem value={"Medicines"}>Medicines</MenuItem>
												</Select>
											</FormControl>
										</Box>
									</div>
									<img
										src={LineImg}
										alt='hr'
										style={{ width: "100%", margin: "2% 0" }}
									/>
									<div className='input-box-itn'>
										<label className='create-label-itn'>Item Name</label>
										<input
											className='create-input'
											value={name}
											type='text'
											name='name'
											onChange={(e) => setname(e.target.value)}
											placeholder='Eg. Bread'
										/>
									</div>

									<div className='qty-unit'>
										<div className='qty-subunit'>
											<label className='create-label'>Quantity</label>
											<div className='qty-input-div'>
												<input
													className='create-input-qty'
													value={qty}
													type='number'
													name='qty'
													placeholder='Eg. 1'
													onChange={(e) => setQty(e.target.value)}
												/>
											</div>
										</div>

										<div className='unit'>
											<label className='create-label-unit'>Units</label>
											{/*<select
												value={unit}
												onChange={(e) => setUnit(e.target.value)}>
												<option value='1'>1</option>
												<option value='2'>2</option>
												<option value='3'>3</option>
											</select> */}
											<div className='units-qty'>
												<Box sx={{ minWidth: 120 }}>
													<FormControl fullWidth size='small' margin='none'>
														<InputLabel
															id='demo-simple-select-label'
															//style={{ padding: '1px' }}
														>
															Unit
														</InputLabel>
														<Select
															labelId='demo-simple-select-label'
															id='demo-simple-select'
															value={unit}
															label='category'
															onChange={(e) => setUnit(e.target.value)}>
															<MenuItem value={"pieces"}>Pieces</MenuItem>
															<MenuItem value={"Kg"}>Kg</MenuItem>
															<MenuItem value={"Ltr"}>Ltr</MenuItem>
														</Select>
													</FormControl>
												</Box>
											</div>
										</div>
									</div>

									{/* <div className='btns'> */}

									<div
										className='add-item-btn'
										onClick={() =>
											addToCart(dispatch, { name, qty, unit })
										}>
										Add Item
									</div>
									{/* <hr className='create-order-hr' /> */}
									<img src={LineImg} alt='hr' style={{ width: "100%" }} />
									<div className='back' onClick={() => setModalVisible(false)}>
										Back
									</div>

									{/* </div> */}
								</div>
							</div>
							<div className='createOrder-right'>
								<div className='right-heads'>
									<span className='label-right'>ORDER NAME: </span>
									<span className='label-right-value'>{orderName}</span>
								</div>
								<div className='right-heads-cat'>
									<span className='label-right'>CATEGORY: </span>
									<span className='label-right-value'>{category}</span>
								</div>
								<div className='items-wrapper'>
									<span className='label-right'>ITEMS IN CART</span>
									<div className='item-cont'>
										{itemsInCart &&
											itemsInCart.map((item) => (
												<>
													<div className='item-row'>
														<span className='label-right-value'>
															{item.name}
														</span>
														<div className='item-row-right'>
															<span className='label-right-value'>
																{item.quantity} {item.unit}
															</span>
															<div>
																<Delete />
																<Edit />
															</div>
														</div>
													</div>		
											</>
										))}
									</div>
								</div>
								<div className='addCart-btn'>
									<div className='checkout-btn'>Check Out</div>
								</div>
							</div>
						</div>
					</div>
				</div>
			) : (
				<div></div>
			)}
		</>
	);
};

export default CreateOrderModal;
