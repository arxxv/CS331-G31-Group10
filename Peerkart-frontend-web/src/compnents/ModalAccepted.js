import React from "react";
import "../css/ModalAccepted.css";

function ModalAccepted({ setModalAcceptedOrderOpen }) {
  return (
    <div className="modalBackground">
      <div className="modalContainer">
        <div className="titleCloseBtn">
          <button
            onClick={() => {
              setModalAcceptedOrderOpen(false);
            }}
          >
            X
          </button>
        </div>
        <div className="title">
          <h1>Accepted Orders</h1>
        </div>
        <div className="acceptbody">
        <div className="accepted_modal_list">
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
              <div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
							<div className="accepted_modal_list_item">
								<span className="accepted_modal_list_item_1 upper">daily needs list</span>
								<span className="accepted_modal_list_item_2 green-color">345</span>
								<span className="accepted_modal_list_item_3 green">View</span>
							</div>
						</div>
					</div>
        </div>
      </div>
  );
}

export default ModalAccepted;
