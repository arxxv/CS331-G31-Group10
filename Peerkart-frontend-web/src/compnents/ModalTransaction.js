import React from "react";
import "../css/ModalTransaction.css";

function ModalTransaction({ setModalTransactionOpen }) {
  return (
    <div className="modalBackground">
      <div className="modalContainer">
        <div className="titleCloseBtn">
          <button
            onClick={() => {
              setModalTransactionOpen(false);
            }}
          >
            X
          </button>
        </div>
        <div className="title">
          <h1>Transaction History</h1>
        </div>
        <div className="transbody">
        <div className="transaction_modal_list">
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
              <div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
              <div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
              <div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
							<div className="transaction_modal_list_item">
								<span className="transaction_modal_list_item_1 upper">daily needs list</span>
								<span className="transaction_modal_list_item_2 green-color">345</span>
								<span className="transaction_modal_list_item_3 green">View</span>
							</div>
						</div>
					</div>
        </div>
      </div>
  );
}

export default ModalTransaction;
