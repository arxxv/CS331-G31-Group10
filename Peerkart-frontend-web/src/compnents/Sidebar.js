import React from "react";
import "../css/Sidebar.css";
import { Link } from "react-router-dom";
import {
	Dashboard,
	CreditCard,
	Login,
	Logout,
	AssignmentInd,
	RocketLaunch,
} from "@mui/icons-material";
import Logo from "../images/logop.png";

const Sidebar = () => {
	return (
		<div className='sidebar-con'>
			<div className='side'>
				<div
					className='logo'
					style={{
						paddingBottom: "20px",
						display: "flex",
						alignItems: "center",
						justifyContent: "center",
						paddingTop: "20px",
					}}>
					<img
						src={Logo}
						style={{
							height: "40px",
							width: "40px",
						}}
					/>
				</div>
				<ul className='side-menu'>
					<Link to='/' className='sidelink'>
						<li className='side-option'>
							<div className='side-icon'>
								<div className='side-icon-shadow'>
									<AssignmentInd />
								</div>
							</div>
							<div className='side-text'>Home</div>
						</li>
					</Link>
					<Link to='/dashboard' className='sidelink'>
						<li className='side-option'>
							<div className='side-icon'>
								<div className='side-icon-shadow'>
									<Dashboard />
								</div>
							</div>
							<div className='side-text'>Dashboard</div>
						</li>
					</Link>
					<Link to='/tracking' className='sidelink'>
						<li className='side-option'>
							<div className='side-icon'>
								<div className='side-icon-shadow'>
									<RocketLaunch />
								</div>
							</div>
							<div className='side-text'>Tracking</div>
						</li>
					</Link>
					<Link to='/login' className='sidelink'>
						<li className='side-option'>
							<div className='side-icon'>
								<div className='side-icon-shadow'>
									<Login />
								</div>
							</div>
							<div className='side-text'>Sign Up</div>
						</li>
					</Link>
				</ul>
			</div>
		</div>
	);
};

export default Sidebar;
