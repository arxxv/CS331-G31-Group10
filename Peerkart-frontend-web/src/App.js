import "./App.css";
import Sidebar from "./compnents/Sidebar";
import Home from "./pages/Home";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import Login from "./pages/Login";
import Register from "./pages/Register";
import Dashboard from "./pages/Dashboard";

import { Provider, useSelector } from "react-redux";

function App() {
	const token=useSelector(state=>state.auth.userData.token);
	return (
		<div>
			<Router>
				<Routes>
					<Route path='/login' element={<Login />} />
					<Route path="/register" element={<Register/>}/>
				</Routes>
				{token && <div className='app'>
					<div className='app-left'>
						<Sidebar />
					</div>

					<div className='app-right'>
						<Routes>
							<Route path='/' element={<Home />} />
							<Route path='/dashboard' element={<Dashboard />} />
						</Routes>
					</div>
				</div>}
			</Router>
		</div>
	);
}

export default App;
