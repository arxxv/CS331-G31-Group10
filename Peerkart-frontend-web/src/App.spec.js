import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from "./redux/store";
import App from './App';
import Login from "./pages/Login";
import Home from "./pages/Home";
import Register from "./pages/Register";
import Dashboard from "./pages/Dashboard";
import {BrowserRouter as Router} from "react-router-dom";
import Sidebar from './compnents/Sidebar';
import CreateOrderModal from './compnents/CreateOrderModal';
import { Modal } from '@mui/material';
import ModalAccepted from './compnents/ModalAccepted';
import ModalTransaction from './compnents/ModalTransaction';
import ProfileDetail from './compnents/ProfileDetails';

describe('App', () => {
	it('should render the site', () => {
		const container = shallow(<Provider store={store}>
            <App />
          </Provider>);
		expect(container).toMatchSnapshot();
});
});

describe('Login', () => {
	it('Login page is rendered', () => {
		const container = shallow(<Router><Login /></Router>);
		expect(container).toMatchSnapshot();
});
});

describe('Register', () => {
	it('Register page is rendered', () => {
		const container = shallow(<Router><Register /></Router>);
		expect(container).toMatchSnapshot();
});
});

describe('Home', () => {
	it('Home page is rendered', () => {
		const container = shallow(<Router><Home /></Router>);
		expect(container).toMatchSnapshot();
});
});

describe('Dashboard', () => {
	it('Dashboard page is rendered', () => {
		const container = shallow(<Router><Dashboard /></Router>);
		expect(container).toMatchSnapshot();
});
});

describe('Sidebar', () => {
	it('Sidebar component is rendered', () => {
		const container = shallow(<Sidebar />);
		expect(container).toMatchSnapshot();
});
});

describe('CreateOrderModal', () => {
	it('CreateOrderModal component is rendered', () => {
		const container = shallow(<Provider store={store}><CreateOrderModal /></Provider>);
		expect(container).toMatchSnapshot();
});
});

describe('Modal', () => {
	it('Modal component is rendered', () => {
		const container = shallow(<Provider store={store}><Modal /></Provider>);
		expect(container).toMatchSnapshot();
});
});

describe('ModalAccepted', () => {
	it('ModalAccepted component is rendered', () => {
		const container = shallow(<Provider store={store}><ModalAccepted /></Provider>);
		expect(container).toMatchSnapshot();
});
});

describe('ModalTransaction', () => {
	it('ModalTransaction component is rendered', () => {
		const container = shallow(<Provider store={store}><ModalTransaction /></Provider>);
		expect(container).toMatchSnapshot();
});
});


describe('ProfileDetail', () => {
	it('ProfileDetail component is rendered', () => {
		const container = shallow(<Provider store={store}><ProfileDetail /></Provider>);
		expect(container).toMatchSnapshot();
});
});