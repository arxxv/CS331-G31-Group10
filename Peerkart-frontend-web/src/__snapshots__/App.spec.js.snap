// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the site 1`] = `ShallowWrapper {}`;

exports[`CreateOrderModal CreateOrderModal component is rendered 1`] = `ShallowWrapper {}`;

exports[`Dashboard Dashboard page is rendered 1`] = `ShallowWrapper {}`;

exports[`Home Home page is rendered 1`] = `ShallowWrapper {}`;

exports[`Login Login page is rendered 1`] = `ShallowWrapper {}`;

exports[`Modal Modal component is rendered 1`] = `ShallowWrapper {}`;

exports[`ModalAccepted ModalAccepted component is rendered 1`] = `ShallowWrapper {}`;

exports[`ModalTransaction ModalTransaction component is rendered 1`] = `ShallowWrapper {}`;

exports[`ProfileDetail ProfileDetail component is rendered 1`] = `ShallowWrapper {}`;

exports[`Register Register page is rendered 1`] = `ShallowWrapper {}`;

exports[`Sidebar Sidebar component is rendered 1`] = `ShallowWrapper {}`;
